python/fledge/plugins/south/person_detection/person_detection.py

```diff
@@ -509,19 +590,27 @@ def run(self):
                     'timestamp': utils.local_timestamp(),
                     'readings': reads
                 }
+
                 async_ingest.ingest_callback(c_callback, c_ingest_ref, data)
 
                 # show the frame on the window
                 try:
                     if self.enable_window:
                         cv2.imshow("Human Detector", frame)
+
                     WebStream.FRAME = frame.copy()
+
                 except Exception as e:
                     _LOGGER.info('exception  {}'.format(e))
 
                 # wait for 1 milli second
                 cv2.waitKey(1)
 
+        _LOGGER.debug("Stopping the stream ")
         self.videostream.stop()
+        _LOGGER.info("Camera stream has been stopped")
         time.sleep(2)
         cv2.destroyAllWindows()
+        _LOGGER.debug("All windows destroyed")
+        WebStream.SHUTDOWN_IN_PROGRESS = True
```

Comment:
Should this flag be set few lines above, say at line 609/610? Think about the worst case of interaction with the code that reads this flag?

Reply:
Done